lib/ecto/type.ex

```diff
@@ -1041,4 +1047,10 @@ defmodule Ecto.Type do
     do: :error
   defp validate_decimal(other),
     do: other
+
+  defp truncate_usec(nil), do: nil
+  defp truncate_usec(struct), do: %{struct | microsecond: {0, 0}}
```

Comment:
We should probably do this only if `microsecond` is not already `{0, 0}`, this can save us some memory allocation.